### PR TITLE
Vendoring and some refactoring

### PR DESCRIPTION
**Pull request by innwkd**

### CODE REVIEW

File: packages/api/smart.go [hunk 1]
```diff
@@ -79,7 +79,6 @@ func validateSmartContract(r *http.Request, data *apiData, result *prepareResult
 					}
 					if !found {
 						log.WithFields(log.Fields{"type": consts.NotFound, "signature": ret[1]}).Error("unknown signature")
-						err = fmt.Errorf(`%s is unknown signature`, ret[1])
```
Comment (gentee) [inline packages/api/smart.go:82]: We cannot delete this err. It must be returned. Maybe to use 
return contract, nil, fmt.Errorf(`%s is unknown signature`, ret[1])
The best way is to add E_UNKNOWNSIGN in error.go and return 
return contract, ret[1], fmt.Errorf(`E_UNKNOWNSIGN`)

File: packages/smart/funcs.go [hunk 2]
```diff
@@ -789,11 +798,12 @@ func CreateColumn(sc *SmartContract, tableName, name, coltype, permissions, inde
 		log.WithFields(log.Fields{"type": consts.JSONUnmarshallError, "error": err}).Error("unmarshalling columns to json")
 		return err
 	}
-	_, _, err = sc.selectiveLoggingAndUpd([]string{`columns`}, []interface{}{string(permout)},
+	sc.selectiveLoggingAndUpd([]string{`columns`}, []interface{}{string(permout)},
```
Comment (gentee) [inline packages/smart/funcs.go]: I think 
return err 
should be here

File: packages/smart/smart_p.go [hunk 3]
```diff
@@ -664,21 +666,23 @@ func RollbackEcosystem(sc *SmartContract) error {
...
 }
 
+// RollbackTable is rolling back table
 func RollbackTable(sc *SmartContract, name string) error {
 	if sc.TxContract.Name != `@1NewTable` {
 		log.WithFields(log.Fields{"type": consts.IncorrectCallingContract}).Error("RollbackTable can be only called from @1NewTable")
 		return fmt.Errorf(`RollbackTable can be only called from @1NewTable`)
 	}
-	err := model.DropTable(sc.DbTransaction, fmt.Sprintf("%d_%s", sc.TxSmart.EcosystemID, name))
+	model.DropTable(sc.DbTransaction, fmt.Sprintf("%d_%s", sc.TxSmart.EcosystemID, name))
```
Comment (gentee) [inline packages/smart/smart_p.go]: I think
if err != nil {
		log.WithFields(log.Fields{"type": consts.DBError, "error": err}).Error("deleting table")
		return err
	}
should be after DropTable

File: packages/config/config.go [hunk 4]
```diff
@@ -27,35 +29,38 @@ type DBConfig struct {
...
 }
 
+// Save is saving config to file
 func Save(logLevel, installType string, dbConf *DBConfig) error {
 	path := *utils.Dir + "/" + configFileName
 	if !IsExist() {
 		ioutil.WriteFile(path, []byte(``), 0644)
 	}
-	confIni, err := config.NewConfig("ini", path)
+	confIni, _ := config.NewConfig("ini", path)
```
Comment (potehinre) [inline packages/config/config.go]: It would be better, not to ignore, but log and return error

File: packages/parser/common_tx_parser.go [hunk 5]
```diff
@@ -141,9 +141,9 @@ func (p *Parser) DeleteQueueTx(hash []byte) error {
 // AllTxParser parses new transactions
 func (p *Parser) AllTxParser() error {
 	logger := p.GetLogger()
-	all, err := model.GetAllUnverifiedAndUnusedTransactions()
+	all, _ := model.GetAllUnverifiedAndUnusedTransactions()
```
Comment (potehinre) [inline packages/parser/common_tx_parser.go]: again, ignoring errors are bad

File: packages/smart/funcs.go [hunk 6]
```diff
@@ -701,7 +709,7 @@ func ColumnCondition(sc *SmartContract, tableName, name, coltype, permissions, i
 	if isExist {
 		return sc.AccessTable(tblName, `update`)
 	}
-	count, err := model.GetColumnCount(tblName)
+	count, _ := model.GetColumnCount(tblName)
```
Comment (potehinre) [inline packages/smart/funcs.go]: ignoring errors are bad

File: packages/smart/smart_p.go (hunk 3 above)
Comment (potehinre) [reply packages/smart/smart_p.go]: i agree

**Review by potehinre — CHANGES_REQUESTED** [review]

File: packages/smart/funcs.go [hunk 7]
```diff
@@ -709,7 +709,10 @@ func ColumnCondition(sc *SmartContract, tableName, name, coltype, permissions, i
 	if isExist {
 		return sc.AccessTable(tblName, `update`)
 	}
-	count, _ := model.GetColumnCount(tblName)
+	count, err := model.GetColumnCount(tblName)
+	if err != nil {
+		return err
```
Comment (potehinre) [inline packages/smart/funcs.go]: You must log it here, because there is a file and line fields in logs, which should point to this location

**Review by potehinre — CHANGES_REQUESTED** [review]

**Pull request merged**